### PR TITLE
Updated LED functions

### PR DESCRIPTION
**Pull request by Atomic-gong**
LED component will now take input through 3 functions (pulse, flash,
solid) which if not provided with colour will continue to use the
current led colour. The values of the leds when using these modes (pulse
brightness, toggle leds in flash) are called in execute.

### CODE REVIEW

File: controllers/leds.py [hunk 1]
```diff
@@ -3,30 +3,36 @@
...
-    leds: StatusLights
+    status_lights: StatusLights
+    vision: Vision
 
     def execute(self):
-        if not self.indexer.has_cargo_in_chimney() and not self.leds.is_flashing:
-            self.leds.flash(LedColours.ORANGE)
+        if not self.vision.is_ready():
+            if not self.status_lights.is_flashing:
+                self.status_lights.flash(LedColours.PINK.value)
```
Comment (auscompgeek) [inline controllers/leds.py:20]: The controller shouldn't care that a colour has a value.
```suggestion
                self.status_lights.flash(LedColours.PINK)
```

File: controllers/leds.py [hunk 2]
```diff
@@ -3,30 +3,36 @@
...
+        elif not self.indexer.has_cargo_in_chimney():
+            if not self.status_lights.is_flashing:
+                self.status_lights.flash(LedColours.ORANGE.value)
```
Comment (auscompgeek) [inline controllers/leds.py:23]:
```suggestion
                self.status_lights.flash(LedColours.ORANGE)
```

File: controllers/leds.py [hunk 3]
```diff
@@ -3,30 +3,36 @@
...
         elif (
-            self.distance > self.shooter_control.MAX_DIST
-            or self.distance < self.shooter_control.MIN_DIST
+            self.shooter_control.distance > self.shooter_control.MAX_DIST
+            or self.shooter_control.distance < self.shooter_control.MIN_DIST
         ):
-            self.status_lights.solid(LedColours.BLUE)
+            self.status_lights.solid(LedColours.BLUE.value)
```
Comment (auscompgeek) [inline controllers/leds.py:28]:
```suggestion
            self.status_lights.solid(LedColours.BLUE)
```

File: controllers/leds.py [hunk 4]
```diff
@@ -3,30 +3,36 @@
...
         elif (
             self.chassis.translation_velocity.norm() > self.shooter_control.MAX_SPEED
             or self.chassis.rotation_velocity.radians()
             > self.shooter_control.MAX_ROTATION
         ):
-            self.status_lights.solid(LedColours.RED)
+            self.status_lights.solid(LedColours.RED.value)
```
Comment (auscompgeek) [inline controllers/leds.py:34]:
```suggestion
            self.status_lights.solid(LedColours.RED)
```

File: controllers/leds.py [hunk 5]
```diff
@@ -3,30 +3,36 @@
...
         # elif not self.shooter.is_at_speed() or not self.turret.is_on_target():
         #     self.status_lights.solid(LedColours.PINK)
         else:
-            self.status_lights.solid(LedColours.GREEN)
+            self.status_lights.solid(LedColours.GREEN.value)
```
Comment (auscompgeek) [inline controllers/leds.py:38]:
```suggestion
            self.status_lights.solid(LedColours.GREEN)
```

**Review by LucienMorey — APPROVED** [review]

**Pull request merged**